index.bs

```diff
@@ -303,6 +284,61 @@ Loader instances are initially created with the internal slots described in the
 
 <h3 id="registry-abstract-operations">Abstract Operations for Registry Objects</h3>
 
+<h4 id="registry-GetCurrentStage" aoid="GetCurrentStage">GetCurrentStage(entry)</h4>
+
+The abstract operation GetCurrentStage with argument entry performs the following steps:
+
+<emu-alg>
+1. If Type(_entry_) is not Object, throw a *TypeError* exception.
+1. Let _stages_ be _entry_.[[Pipeline]].
+1. Return the first element of _stages_.
+</emu-alg>
+
+<h4 id="registry-GetStage" aoid="GetStage">GetStage(entry, stage)</h4>
+
+The abstract operation GetStage with arguments entry and stage performs the following steps:
+
+<emu-alg>
+1. If Type(_entry_) is not Object, throw a *TypeError* exception.
+1. Let _stages_ be _entry_.[[Pipeline]].
+1. For each element _entry_ of _stages_, do
+  1. If _entry_.[[Stage]] is equal to _stage_, then
+    1. Return _entry_.
+1. Return *undefined*.
+</emu-alg>
+
+<h4 id="registry-UpgradeToStage" aoid="UpgradeToStage">UpgradeToStage(entry, stage)</h4>
```

Comment:
this is the important key in this PR. With `UpgradeToStage()` we can simply drop old stages from the pipeline when reaching new stages, we don't need to worry about removing the last one, the "ready" stage will always we preserved, all other stages will be collected.